pyccel/ast/basic.py

```diff
@@ -339,11 +349,28 @@ def fst(self):
         ast.AST
             The AST object which was parsed.
         """
-        if len(self._fst) == 1:
-            return self._fst[0]
+        if len(self._ast) == 1:
+            return self._ast[0]
         else:
             return None
 
+    @ast.setter
+    def ast(self, ast_node):
+        if not isinstance(ast_node, ast.AST):
+            raise TypeError(f"ast_node must be an AST object, not {type(ast_node)}")
+
+        if self.ast:
+            if hasattr(ast_node, 'lineno'):
+                if self.ast.lineno != ast_node.lineno or self.ast.col_offset != ast_node.col_offset:
+                    self._ast.append(ast_node)
+        else:
+            if not hasattr(ast_node, 'lineno'):
+                # Handle module object
+                ast_node.lineno     = 1
+                ast_node.col_offset = 1
+
+            self._ast.append(ast_node)
```

Comment:
I find it strange that the property getter returns `self._ast[0]` while the setter appends to the list. If the list contains only one element there is no problem, but if we assign twice:
```python
parser.ast = ast0
parser.ast = ast1
```
one will discover that `parser.ast is None` which is surprising.

This behavior is inherited from the previous code which used the `set_fst` function. Maybe we could explain in the docstrings what is going on now?

Reply:
Ah! I guess that explains why I wasn't using a setter before

Reply:
Fixed: 3d9edd00f
Unless you would rather I went back to `set_ast`? Or even `set_current_ast`?

Reply:
Fixed 7aafa4b81

Reply:
Excellent description!